src/ArborX_DBSCAN.hpp

```diff
@@ -374,8 +374,8 @@ dbscan(ExecutionSpace const &exec_space, Primitives const &primitives,
           Details::FDBSCANDenseBoxCallback<MemorySpace, CorePoints, Primitives,
                                            decltype(dense_cell_offsets),
                                            decltype(permute)>{
-              labels, CorePoints{}, primitives, dense_cell_offsets, permute,
-              eps});
+              labels, CorePoints{}, primitives, dense_cell_offsets, exec_space,
+              permute, eps});
```

Comment:
I was a little surprised that we don't pass the execution space as the first argument.